### PR TITLE
checkcommits static checks

### PR DESCRIPTION
**Pull request by grahamwhaley**
Fix up some items reported by the go static checkers.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Hi @jodh-intel 
I've marked this as DNM, as there is another static check report that could do with fixing that is not in this series....
gocyclo (as invoked like https://github.com/clearcontainers/runtime/blob/master/.ci/go-static-checks.sh#L92) is complaining that the comlexity of getCommitAndBranch is '16', and we have the limit set to '15' :-(
I had a look at the function, and it's not a bad function, just happens to be full of a pretty linear series of checks. Any ideas on if there is a nice way to reduce the complexity of that, without somewhat artificially slicing it up?

And, of course, whilst here your input on the commits would be welcome - particularly some of the simplification ones, which whilst I have no doubt are correct, I'd like your input on if they do improve readability/clarify, or if they actually detract something.

**Comment by grahamwhaley** [comment]
Updated. Thanks to @jodh-intel for pushing #106 - meant I could proceed and find the remaining items.
@GabyCT you might want to check the fetchbranches related ones (minor stuff)

**Comment by jodh-intel** [comment]
Hi @grahamwhaley - looks like you've hit the Travis bug again, so it's another close-and-reopen-fresh-PR I'm afraid ;(

**Comment by grahamwhaley** [comment]
@jodh-intel I think we are actually hitting a different sort of Travis issue. I did a minor commit message edit and re-pushed and now it has passed. Similar to the rebase I did yesterday to fix the same problem on a different PR.

File: cmd/checkcommits/checkcommits_test.go [hunk 1]
```diff
@@ -642,7 +642,7 @@ func TestGetCommitAndBranch(t *testing.T) {
 		}
 
 		// XXX: crucially, no arguments (to trigger the auto-detection)
-		commit, dstBranch, err := getCommitAndBranch([]string{}, []string{})
+		commit, dstBranch, _ := getCommitAndBranch([]string{}, []string{})
```
Comment (jodh-intel) [inline cmd/checkcommits/checkcommits_test.go]: Sorry, just noticed this: this is an oversight - we should be retrieving the error, then:

```
if err != nil {
    t.Fatal(err)
}
```

File: cmd/fetchbranches/main.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ func main() {
 	fmt.Printf("PULL_REQUEST_NUMBER : %d, OWNER : %s, REPOSITORY: %s\n", pr.number, pr.owner, pr.repo)
 
 	// Shows author PR comments
-	comments, err := pr.GetAuthorComments()
+	comments, _ := pr.GetAuthorComments()
```
Comment (jodh-intel) [inline cmd/fetchbranches/main.go]: Same as above - I think the original code was just missing a check on `err`.

File: cmd/checkcommits/checkcommits_test.go (hunk 1 above)
Comment (grahamwhaley) [inline cmd/checkcommits/checkcommits_test.go]: ok, np, I'll rework. Yep, I did just pretty blindly apply the 'is ignored' ruling. Must remember to engage brain a little more next time

Comment (grahamwhaley) [inline cmd/checkcommits/checkcommits_test.go]: updated and re-pushed

**Comment by grahamwhaley** [comment]
Rebased and pushed.
Also noted that I'd broken 'go test' in fetchbranches - now also fixed.

**Comment by devimc** [comment]
lgtm

**Pull request merged**